test/react-native/features/fixtures/app/scenarios/SessionJsControlledManualJsScenario.js

```diff
@@ -0,0 +1,39 @@
+import Scenario from './Scenario'
+import Bugsnag from '@bugsnag/react-native'
+
+export class SessionJsControlledManualJsScenario extends Scenario {
+  constructor(configuration, extraData, jsConfig) {
+    super()
+    configuration.autoTrackSessions = false
+  }
+  run() {
+    Bugsnag.startSession()
+    setTimeout(() => {
+      Bugsnag.notify(new Error('SessionJsControlledManualJsScenarioA'), () => {}, () => {
+        setTimeout(() => {
+          Bugsnag.pauseSession()
+          setTimeout(() => {
+            Bugsnag.notify(new Error('SessionJsControlledManualJsScenarioB'), () => {}, () => {
+              setTimeout(() => {
+                Bugsnag.resumeSession()
+                setTimeout(() => {
+                  Bugsnag.notify(new Error('SessionJsControlledManualJsScenarioC'), () => {}, () => {
+                    setTimeout(() => {
+                      Bugsnag.pauseSession()
+                      setTimeout(() => {
+                        Bugsnag.startSession()
+                        setTimeout(() => {
+                          Bugsnag.notify(new Error('SessionJsControlledManualJsScenarioD'))
```

Comment:
if you promisify setTimeout, this pyramid of doom would go away

e.g. https://stackoverflow.com/a/33292942

Reply:
Hopefully that's been resolved now, pending CI